### PR TITLE
SPS Project Reviews

### PR DESCRIPTION
**Pull request by yyzx117**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,6 +1,6 @@
 # Google Software Product Sprint
 
-This repo contains the projects you'll work on throughout SPS.
+This repo contains Zhixuan Yang's portfolio and SPS projects.
```
Comment (bolesgb) [inline README.md:3]: Could you mind breaking this up into multiple prs? that way it's a bit easier to review?

**Comment by bolesgb** [comment]
Another good practice is to have a descriptive title and description to your prs. So right now it says `
Update the README.md file to describe the repo` but that's not what the pr is doing.

**Comment by bolesgb** [comment]
Also it's always a good idea for front end changes to include a screenshot of the change you're making to the front end so it's more visible to the reviewer.  This can be added to the description section of the pr

**Review by bolesgb — APPROVED** [review]

**Review by yyzx117 — COMMENTED** [review]
Wrong comment.

File: portfolio/src/main/java/com/google/sps/servlets/HelloWorldServlet.java [hunk 2]
```diff
@@ -9,10 +9,14 @@
...
+    private int pageViews = 0;
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    pageViews++;
+
     response.setContentType("text/html;");
-    response.getWriter().println("<h1>Hello world!</h1>");
+    response.getWriter().println("<h1>Hello Zhixuan!</h1>");
+    response.getWriter().println("<p>This page has been viewed " + pageViews + " times.</p>");
```
Comment (yyzx117) [inline portfolio/src/main/java/com/google/sps/servlets/HelloWorldServlet.java:20]: I added a counter to show how many times the page has been viewed.

File: portfolio/src/main/java/com/google/sps/servlets/HelloWorldServlet.java [hunk 3]
```diff
@@ -9,10 +9,14 @@
...
 public class HelloWorldServlet extends HttpServlet {
+    private int pageViews = 0;
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    pageViews++;
+
     response.setContentType("text/html;");
-    response.getWriter().println("<h1>Hello world!</h1>");
+    response.getWriter().println("<h1>Hello Zhixuan!</h1>");
```
Comment (yyzx117) [inline portfolio/src/main/java/com/google/sps/servlets/HelloWorldServlet.java:19]: Just by tutorial.

File: portfolio/src/main/java/com/google/sps/servlets/FormHandlerServlet.java [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+    // Get the value entered in the form.
+    String textValue = request.getParameter("text-input");
+    boolean phone = Boolean.parseBoolean(request.getParameter("phone"));
+    boolean email = Boolean.parseBoolean(request.getParameter("email"));
+
+    // Write the value to the response so the user can see it.
+    response.getWriter().println("Thanks for your submission!");
+
+    if (phone) {
+        System.out.println("Your Phone number: " + textValue);
```
Comment (yyzx117) [inline portfolio/src/main/java/com/google/sps/servlets/FormHandlerServlet.java:24]: Implement the options and write the message to the response.

File: portfolio/src/main/java/com/google/sps/servlets/FormHandlerServlet.java [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+
+    // Get the value entered in the form.
+    String textValue = request.getParameter("text-input");
+    boolean phone = Boolean.parseBoolean(request.getParameter("phone"));
+    boolean email = Boolean.parseBoolean(request.getParameter("email"));
+
+    // Write the value to the response so the user can see it.
+    response.getWriter().println("Thanks for your submission!");
+
```
Comment (yyzx117) [inline portfolio/src/main/java/com/google/sps/servlets/FormHandlerServlet.java:22]: Implement the options and write the message to the response.

File: portfolio/src/main/webapp/style.css [hunk 6]
```diff
@@ -182,4 +190,11 @@ h1 {
     position: relative;
     top: 400px;
     height:100px;
+}
```
Comment (yyzx117) [inline portfolio/src/main/webapp/style.css]: Design the position of the POST request below.

File: portfolio/src/main/webapp/style.css [hunk 7]
```diff
@@ -4,6 +4,7 @@
   width: 650px;
 }
```
Comment (yyzx117) [inline portfolio/src/main/webapp/style.css:5]: Simply add some comments that you could ignore.

File: portfolio/src/main/java/com/google/sps/servlets/FormHandlerServlet.java (hunk 5 above)
Comment (yyzx117) [reply portfolio/src/main/java/com/google/sps/servlets/FormHandlerServlet.java:22]: Add function of changing lines.

File: portfolio/src/main/webapp/index.html [hunk 8]
```diff
@@ -8,79 +8,101 @@
...
+            <p id='para11'> Volleyball</p>
+            <p id='para12'> Food </p>
+            <p id='para13'> Singing </p>
+            <p id='random'>Learn about my favourite Singers by clicking the button below:</p>
+            <button id='button' onclick="addRandomGreeting()"> Click me </button>
+            <div id="greeting-container"></div>
+            <div class="slidecontainer">
+            <input type="range" min="0" max="4" value="0" class="slider" id="myRange">
+            </div>
+            <p id='demo'></p>
```
Comment (yyzx117) [inline portfolio/src/main/webapp/index.html]: Add a slider that I want to implement into pictures later.

File: portfolio/src/main/webapp/index.html [hunk 9]
```diff
@@ -8,79 +8,101 @@
   </head>
```
Comment (yyzx117) [inline portfolio/src/main/webapp/index.html:8]: Just redesign the whole page.

File: portfolio/src/main/webapp/style.css [hunk 10]
```diff
@@ -1,7 +1,14 @@
+body {
```
Comment (yyzx117) [inline portfolio/src/main/webapp/style.css:1]: Implement the gradient color background.

File: portfolio/src/main/webapp/script.js [hunk 11]
```diff
@@ -12,6 +12,15 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
```
Comment (yyzx117) [inline portfolio/src/main/webapp/script.js:14]: Implement the slider.

File: portfolio/src/main/webapp/index.html [hunk 12]
```diff
@@ -26,18 +26,51 @@ <h1>Zhixuan Yang</h1>
...
+            <br>
+            <!-- The dots/circles -->
+            <div class='dots' style="text-align:center">
+                <span class="dot" onclick="currentSlide(1)"></span>
+                <span class="dot" onclick="currentSlide(2)"></span>
+                <span class="dot" onclick="currentSlide(3)"></span>
+                <span class="dot" onclick="currentSlide(4)"></span>
+            </div>
+
+
```
Comment (yyzx117) [inline portfolio/src/main/webapp/index.html:76]: image

File: portfolio/src/main/webapp/index.html [hunk 13]
```diff
@@ -15,7 +15,8 @@ <h1>Zhixuan Yang</h1>
         <p id='contact'> <a id='github' href='https://github.com/yyzx117/software-product-sprint.git'> Github <a/> | <a id='linkedin' href='https://www.linkedin.com/in/zhixuan-yang-7a02b61b9/'>Linkedin</a></p>
 
         <div class='intro'>
-            <p id='intro'> Hi! My name is Zhixuan Yang and I am a junior at Georgia Tech 🐝 majoring in Computer Science and minoring in Psychology.
+            <p id='intro'> Hi! My name is Zhixuan Yang and I am a junior at Georgia Tech 
+                <a href='https://www.gatech.edu/' style="text-decoration: none"><img id='buzz' src='./images/buzz.png'/> </a> majoring in Computer Science and minoring in Psychology.
             Full of passion for tech and life, I aim at becoming a brilliant software engineer who could
```
Comment (yyzx117) [inline portfolio/src/main/webapp/index.html:30]: image

File: portfolio/src/main/java/com/google/sps/servlets/HelloWorldServlet.java (hunk 2 above)
Comment (yyzx117) [reply portfolio/src/main/java/com/google/sps/servlets/HelloWorldServlet.java:20]: image

File: portfolio/src/main/java/com/google/sps/servlets/FormHandlerServlet.java (hunk 5 above)
Comment (yyzx117) [reply portfolio/src/main/java/com/google/sps/servlets/FormHandlerServlet.java:22]: image

File: portfolio/src/main/webapp/index.html [hunk 14]
```diff
@@ -54,5 +54,25 @@ <h1>Zhixuan Yang</h1>
...
+        </label>
+        <br/>
+        <label>
+            <input type="checkbox" name="email" value="true">
+            Email Address
+        </label>
+        <br/><br/>
+        <input type="submit" />
+      </form>
+
```
Comment (yyzx117) [inline portfolio/src/main/webapp/index.html:152]: image

Comment (yyzx117) [reply portfolio/src/main/webapp/index.html:152]: image

File: portfolio/src/main/java/com/google/sps/servlets/FormHandlerServlet.java (hunk 5 above)
Comment (yyzx117) [reply portfolio/src/main/java/com/google/sps/servlets/FormHandlerServlet.java:22]: image

File: portfolio/src/main/webapp/index.html [hunk 15]
```diff
@@ -12,7 +12,17 @@
...
+            <a id='github' href='https://github.com/yyzx117/software-product-sprint.git' style="text-decoration: none"> 
+                <img id='github' src='./images/GitHub.png'/>
+            <a/>
+            <a id='linkedin' href='https://www.linkedin.com/in/zhixuan-yang-7a02b61b9/' style="text-decoration: none">
+                <img id='linkedin' src='./images/LinkedIn.png'/>
+            </a>
+            <a id='email' href = "mailto: yangzhixuan18@gmail.com" style="text-decoration: none">
+                <img id='email' src='./images/email.jpg'/>
+            </a>
+        </p>
```
Comment (yyzx117) [inline portfolio/src/main/webapp/index.html:26]: image

File: portfolio/src/main/webapp/style.css [hunk 16]
```diff
@@ -345,6 +357,11 @@ h1 {
     position:absolute;
     top: 40px;
 }
+#getDate {
+    position: absolute;
+    top: 25px;
+    left: 150px;
+}
```
Comment (yyzx117) [inline portfolio/src/main/webapp/style.css:364]: image

File: portfolio/src/main/java/com/google/sps/servlets/ServerStats.java [hunk 17]
```diff
@@ -0,0 +1,39 @@
...
+
+        // Convert the server stats to JSON
+        Gson gson = new Gson();
+        String json = gson.toJson(arr);
+
+        // Send the JSON as the response
+        response.setContentType("application/json;");
+        response.getWriter().println(json);
+    }
+}
```
Comment (yyzx117) [inline portfolio/src/main/java/com/google/sps/servlets/ServerStats.java:39]: image

File: portfolio/src/main/webapp/index.html [hunk 18]
```diff
@@ -140,6 +140,10 @@ <h1>Zhixuan Yang</h1>
                 <p id="getDate"></p>
             </div>
             <p id='server'>Click <a id='here'href="/hello">here</a> to see content created by server code.</p>
+            <p id='city'>Click <a id='here' href="/server-stats">here</a> to see all the cities I have been to in USA.</p>
+            <p id='random2'>Click this button to see the random city I have been to in USA.</p>
+            <button id='showCity' onclick="getServerStats()">Explore cities</button>
+            <ul id="server-stats-container"></ul>            
         </div>
```
Comment (yyzx117) [inline portfolio/src/main/webapp/index.html:147]: image

File: portfolio/src/main/java/com/google/sps/servlets/ServerStats.java [hunk 19]
```diff
@@ -0,0 +1,39 @@
...
+        arr.add(str6);
+
+        // Convert the server stats to JSON
+        Gson gson = new Gson();
+        String json = gson.toJson(arr);
+
+        // Send the JSON as the response
+        response.setContentType("application/json;");
+        response.getWriter().println(json);
+    }
```
Comment (yyzx117) [inline portfolio/src/main/java/com/google/sps/servlets/ServerStats.java:38]: image

**Pull request merged**